reproducible/nix-build.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+set -x
+. /home/$USER/.nix-profile/etc/profile.d/nix.sh
+
+if [ $OE_SIMULATION ]
+then 
+OE_SIM="--argstr OE_SIM OE_SIMULATION=1"
+fi
+
+nix-build -I. shell.nix --substituters 'https://cache.nixos.org' \
+	    --argstr REV $BUILD_REV \
+	    --argstr SHA $BUILD_SHA \
+	    --arg DO_CHECK $DO_CHECK ${OE_SIM}
+#
+# We have built the nix environment, we need to export it 
+# so we can use it in other projects. 
+nix-store --dump-db >/output/oe-nix-store-$(date +%Y%j%H%M).nixdump
+
+nix-shell -I. shell.nix --substituters 'https://cache.nixos.org' \
```

Comment:
What does this step do, beyond what's done on line 10?

Reply:
We use patchelf to remove the nix-oriented rpath and ld interpreter.  When we do that it makes it difficult to build further projects using the nix store.  We save the nix-store before beginning the patchelf process.

We do the ctests, patchelf and packaging because the very severe visibility restrictions in nix-build won't allow ctests to run.

Reply:
"difficult to build further projects using the nix store" - what are the "further projects"? The OE test apps run in the ctest stage?

Reply:
"use patchelf to remove the nix-oriented rpath and ld interpreter" discussion should be included included in the design doc or readme.

Reply:
I beleive it is.  I will reread and ensure it is sufficiently prominent.

Reply:
If there are downstream projects which would like to build in nix, we can accommodate them.

Reply:
Take a look at ctest_openenclave.sh for an example. We export the build closure before we adjust the ld interpreter and rpath to take the binaries out of nix-world.  The closures retain their nix features.